docs/man/sat-jobstat.8.rst

```diff
@@ -0,0 +1,61 @@
+=============
+ SAT-JOBSTAT
+=============
+
+-------------------------
+Jobstat Short Description
+-------------------------
+
+:Author: Hewlett Packard Enterprise Development LP.
+:Copyright: Copyright 2022 Hewlett Packard Enterprise Development LP.
+:Manual section: 8
+
+SYNOPSIS
+========
+
+**sat** [global-opts] **jobstat** [options]
+
+DESCRIPTION
+===========
+
+Jobstat is a command that allows a user or admin to access application and job data via the command line. The command provides a table showing the user the apid, jobid, user, state, and time-reported of all jobs on the system.
+
+OPTIONS
+=======
+
+These options must be specified after the subcommand.
+
+**-h, --help**
+        Print the help message for 'sat jobstat'.
+
+**-a, --all**
+        Show all application information that is currently documented on the system.
+
+Summarize Remaining Options
+
+EXAMPLES
+========
+
+An example usage of the command:
+
+::
+
+    # sat jobstat
++---------------------------------------+--------------+-------+-------------------------------------+-----------+-----------+---------------------+
+| apid                                  | jobid        | user  | command                             | state     | num_nodes | node_list           |
++---------------------------------------+--------------+-------+-------------------------------------+-----------+-----------+---------------------+
+| 7677b7d3-ec35-4e23-8678-92b823347939  | 49.pbs-host  | 5827  | mpiexec --ppn=1 -n2 ./signals       | completed | 2         | nid000001,nid000002 |
+| 76b783af-5c7e-4462-97c0-3dfd0f04b9b3  | 51.pbs-host  | 5827  | mpiexec --ppn=1 -n2 ./signals       | completed | 2         | nid000001,nid000002 |
+| 7a85c000-05a9-40cd-8551-00092cead801  | 48.pbs-host  | 5827  | mpiexec -n2 ./signals               | completed | 1         | nid000001           |
+| 8b5b14b3-8215-4cbe-bba4-7d7c82d48fef  | 50.pbs-host  | 5827  | mpiexec --ppn=1 -n2 ./signals       | completed | 2         | nid000001,nid000002 |
+| 957bbd4c-de3c-424d-99d6-fd2b728b2941  | 54.pbs-host  | 5827  | mpiexec --ppn=1 -n2 ./signals       | completed | 2         | nid000001,nid000002 |
+| ba977b71-40b6-40db-aa20-a583edc18be3  | 59.pbs-host  | 5827  | mpiexec --ppn=1 -n2 ./signals-sleep | completed | 2         | nid000001,nid000002 |
+| fea2726a-c531-4acd-b158-3fae197999dd  | 57.pbs-host  | 5827  | mpiexec --ppn=1 -n2 ./signals       | completed | 2         | nid000001,nid000002 |
++---------------------------------------+--------------+-------+-------------------------------------+-----------+-----------+---------------------+
```

Comment:
It might be kind of difficult to come up with exact example output, but would be nice to add an example that uses the filter options, e.g. "list all jobs that have completed", which would be done with:

```
# sat jobstat --filter state=completed
```

Reply:
This is a request for a new command option.

Reply:
You may have not realized it but it already exists because Elmir added the format and filter options in `add_jobstat_subparser` (see the function on line 30 of `sat/cli/jobstat/parser.py`:

```
def add_jobstat_subparser(subparsers):
    """Add the jobstat parser to the parent parser.
    Args:
        subparsers: The argparse.ArgumentParser object returned by the
            add_subparsers method.
    Returns:
        None
    """
    format_options = sat.parsergroups.create_format_options()
    filter_options = sat.parsergroups.create_filter_options()
```

It hooks into the `Report` class that is being used in `main.py`, so this should already work. Is it possible to test it to verify?

Reply:
I have not gotten to test this yet. Elmir's gave a demo, so know that code works. But I will do a round of testing after addressing review comments.

Reply:
I will make the current example to have the filter and give a more general example later.

Reply:
Testing would be very helpful. In the meantime I can confirm that the format and filter options are in the help text:
```
mbp-ekamin:sat ekamin$ sat jobstat --help
usage: sat jobstat [-h] [--format {pretty,yaml,json}] [--no-borders]
                   [--no-headings] [--reverse] [--sort-by FIELD]
                   [--show-empty] [--show-missing] [--fields FIELDS]
                   [--filter QUERY] [--all]

This command shows all applications and relevent information pertaining to
those applications.

optional arguments:
  -h, --help            show this help message and exit
  --all, -a             Show all application information available on the
                        system.

format options:
  Options to modify output formatting.

  --format {pretty,yaml,json}
                        Display information in the given format. Defaults to
                        'pretty'.
  --no-borders          Omit borders from tables.
  --no-headings         Omit headings from tables.
  --reverse             Sort the output in reverse order. Only applies for
                        "pretty".
  --sort-by FIELD       Select which column to sort by. Can accept a column
                        name or a 0-based index. Enclose the column name in
                        double quotes if it contains a space.
  --show-empty          Show values for columns even if every value is EMPTY.
  --show-missing        Show values for columns even if every value is MISSING
  --fields FIELDS       Display only the given comma-separated list of fields.

filtering options:
  Options to filter output.

  --filter QUERY        Filter rows of the output based on the query provided.
                        Refer to the man page for this subcommand for more
                        details regarding filter query syntax.
```

Regarding testing, I am particularly unsure about whether the URL change works or if you will need additional changes to handle API authentication as I mention in this comment: https://github.com/Cray-HPE/sat/pull/38#discussion_r968926035